File: IDATT2101/oblig/1/task1-1.c
#include <stdio.h>
#include <stdint.h>

#define I32_MIN 1 << 31

int main()
{
    int32_t prices[] = {-1, 3, -9, 2, 2, -1, 2, -1};
    int32_t len = 8;
    int32_t buy, sell, stonk = 0, best = I32_MIN;

    for (int32_t i = 0; i < len; i++) {         /* n */
        stonk = 0;
        for (int32_t j = i; j < len; j++) {     /* n * n */
            stonk += prices[j];
            if (stonk > best) {
                best = stonk;
                buy = i;
                sell = j;
            }
        }
    }

    printf("kjøpsdag: %d, salgsdag: %d, profit: %d\n", buy, sell + 1, best);
    return 0;
}

File: IDATT2101/oblig/1/task1-1.go
package main

import "fmt"

func main() {
    prices := []int{-1, 3, -9, 2, 2, -1, 2, -1}
    var buy, sell, stonk, best int

    for i := 0; i <= len(prices); i++ {
        stonk = 0
        for j := i; j < len(prices); j++ {
            stonk += prices[j]
            if stonk > best {
                best = stonk
                buy = i
                sell = j
            }
        }
    }

    fmt.Printf("kjøpsdag: %d, salgsdag: %d, profit: %d\n", buy, sell + 1, best)
}
